IdentityCore/src/cache/mac/MSIDMacKeychainTokenCache.m

```diff
@@ -870,6 +973,121 @@ - (BOOL)clearWithContext:(id<MSIDRequestContext>)context
     return YES;
 }
 
+#pragma mark - Access Control Lists
+
+- (id)createAccess:(MSIDCacheKey *)key context:(id<MSIDRequestContext>)context error:(__unused NSError **)error
+{
+    if (key.isShared)
+    {
+        if (!self.accessForSharedBlob)
+        {
+            if (!self.trustedApplications)
+            {
+                self.trustedApplications = [self createAccessForCurrentApp:context error:error];
+            }
+            
+            self.accessForSharedBlob =  [self accessCreateWithChangeACL:self.trustedApplications
```

Comment:
What about thread safety?

Reply:
Completely forgot about thread safety.:) Just added a synchronized block around this piece of code . I don't believe we need a serialized queue specifically for this purpose. Let me know if you feel otherwise.

Reply:
If this code is executed on each read operation, it might become a bottleneck. Can we just create access objects in the initialization and let reading operation use self.accessObject directly?